mantidproject/mantid
# Move DataProcessorUI to its own directory

Moved all classes defining DataProcessor UI to a new directory under `MantidWidgets`.

**To test:**
- All tests should pass
- Code review (most changes are related to namespaces and includes, but note that I also had to modify `MantidQt/CustomInterfaces/test/CMakeLists.txt` and `MantidQt/MantidWidgets/test/CMakeLists.txt`to include additional dependencies and test helpers)
- Make sure you agree that `MantidWidgets` is the right location (in the [design document](https://github.com/mantidproject/documents/blob/master/Design/DataProcessorAlgorithmUI/DataProcessingUserInterface.md) it was suggested to move the UI into the MantidQtAPI package, but I think `MantidWidgets` makes more sense, as the `DataProcessorUI` is a new widget that can be added to any interface as any other widget).

Fixes #16413.

Internal change, does not need to be in the release notes.

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
